### PR TITLE
Redux structure

### PR DESCRIPTION
**Pull request by pierr**
- Use combine reducer
- Add an action builder
- Add a reducer builder
- Improve example API
- Add a dumb component
- There is a plain redux example with all `entity` related stuff
- There is an example using `actionBuilder` and `reducerBuilder` with all `user` stuff

## action builder

Action builder is a simple way to create action types, action creator, and an async action
It takes one object parameter
### doc
```javascript
{
  name, //The name of your data in the store (example user)
  type, //Which type of action it is: load, save, delete, ...
  service // An async action to load data
}
```

### example
```javascript
const loadAction = actionBuilder({name:'user', type:'load', service:loadUserSvc});
```

will produce

```javascript
export const loadUserTypes = loadAction.types; 
//which are the action types {REQUEST_LOAD_USER, RECEIVE_LOAD_USER}
export const loadUserAction = loadAction.action; 
//which is a function taking the criteria as param
```

## reducer builder
Build a reducer in order to ease the creation on these in 80% of the cases

### doc

To build a reducer you need to provide an object which is

```javascript
{
 types : {
   load: { request, receive, error}// which are the types for the load action
   save: { request, receive, error} //which are the types for the save action
 },
 defaultState: the default state of your reducer
}
```

### example

```javascript
const DEFAULT_USER_STATE = {firstName: 'Test'};
const userReducer = buildReducer({types: {
   load: {request: REQUEST_LOAD_USER, receive: RECEIVE_LOAD_USER},
   save: {request: REQUEST_SAVE_USER, receive: RECEIVE_SAVE_USER}
 },
 defaultState: DEFAULT_USER_STATE
})
```

will produce

```javascript
function userReducer(state = DEFAULT_STATE, {type, payload}){
const {data} = state;
 switch (type) {
  case REQUEST_LOAD_USER:
      return {data, isLoading: true};
  case RECEIVE_LOAD_USER:
      return {data: payload, isLoading: false};
  default:
      return state
 }
}
```

### CODE REVIEW

File: src/actions/entity-actions-builder.js [hunk 1]
```diff
@@ -1,9 +1,78 @@
...
-  return {name: REQUEST_ENTITY, action: loadAction};
-}
-function buildReceiveAction({name}){
-  const RECEIVE_ENTITY = `RECEIVE_ENTITY_${name}`;
+import {capitalize, toUpper} from 'lodash/string';
+
+// A simple function to create action creators
+// Return a function which returns a type and a payload
+// example:  _baseActionCreator('REQUEST_LOAD_USER') will return `payload => {type: 'REQUEST_LOAD_USER', payload}`
+const _baseActionCreator = type => (payload => ({type, payload}));
```
Comment (Bernardstanislas) [inline src/actions/entity-actions-builder.js]: maybe `_actionCreatorBuilder` would fit better ?

File: src/actions/entity-actions-builder.js [hunk 2]
```diff
@@ -1,9 +1,78 @@
...
+// You have to provide a object with the following properties
+// {
+//   service: A `Promise base service` which take one argument (the service entry data) an optional second the options
+//   creators : {
+//     receive:{name, value} where value is the function standing for an action creator
+//     request:{name, value} where value is the function standing for an action creator
+//     error:{name, value} where value is the function standing for an action creator
+//   }
+// }
+const _baseActionAsync = ({service: promiseSvc, creators:{receive: {value: receiveActionCreator}, request: {value: requestActionCreator}, error: {value: errorActionCreator}}}) => (data => {
```
Comment (Bernardstanislas) [inline src/actions/entity-actions-builder.js]: maybe `_asyncActionCreator` would fit better ?

**Comment by pierr** [comment]
@Bernardstanislas I would like your opinion on the `reducerBuilder` thing.
I'm not sure it is needed.

**Comment by pierr** [comment]
@Tommass I would like your remarks on all this PR...

**Comment by pierr** [comment]
@Bernardstanislas @Tommass I merge this but I would like your remarks.

**Comment by pierr** [comment]
@Bernardstanislas would like to have for `reducerBuilder` a `load.RECEIVE` and I agree.

**Comment by Tommass** [comment]
```
const loadAction = actionBuilder({name:'user', type:'load', service:loadUserSvc});
```

I would have renamed `name` parameter as `store`.

**Comment by pierr** [comment]
@Tommass Name is not the store is should be the name of the entity you are using

**Pull request merged**